<?xml version="1.0"?>
<launch>


<!-- <arg name="meshfile" default="/home/amock/datasets/physics_building/physics.dae" /> -->
<!-- <arg name="meshfile" default="/home/amock/workspaces/ros/mamcl_ws/src/uos_tools/uos_gazebo_worlds/Media/models/avz_neu.dae" /> -->
<arg name="meshfile" default="/home/amock/workspaces/imagine_stack/imagine/dat/many_objects.dae" />

<node pkg="imagine_ros" type="mesh_publisher" name="mesh_publisher" output="screen">
    <param name="file" type="string" value="$(arg meshfile)" />
    <param name="frame" type="string" value="map" />
</node>

<node pkg="imagine_ros" type="lidar_simulator" name="lidar_simulator" output="screen">
    <param name="map_file" type="string" value="$(arg meshfile)" />
    <param name="map_frame" type="string" value="map" />
    <param name="sensor_frame" type="string" value="lidar_link" />
    <rosparam param="Tsb">[0.0, 3.65, 1.0, 0.0, 0.0, 0.0]</rosparam>
</node>

<node pkg="imagine_ros" type="camera_simulator" name="camera_simulator" output="screen">
    <param name="map_file" type="string" value="$(arg meshfile)" />
    <param name="map_frame" type="string" value="map" />
    <param name="sensor_frame" type="string" value="camera_link" />
    <rosparam param="Tsb">[0.0, 0.0, 1.0, 0.0, 0.0, 0.0]</rosparam>
</node>

<node pkg="imagine_ros" type="o1dn_simulator" name="o1dn_simulator" output="screen">
    <param name="map_file" type="string" value="$(arg meshfile)" />
    <param name="map_frame" type="string" value="map" />
    <param name="sensor_frame" type="string" value="o1dn_sensor_link" />
    <rosparam param="Tsb">[0.0, -3.65, 1.0, 0.0, 0.0, 0.0]</rosparam>
</node>

<node pkg="imagine_ros" type="ondn_simulator" name="ondn_simulator" output="screen">
    <param name="map_file" type="string" value="$(arg meshfile)" />
    <param name="map_frame" type="string" value="map" />
    <param name="sensor_frame" type="string" value="ondn_sensor_link" />
    <rosparam param="Tsb">[0.0, -7.3, 1.0, 0.0, 0.0, 0.0]</rosparam>
</node>


</launch>